Understanding swim lanes (#165)
alan-turing-institute/AutisticaCitizenScience: project-management/Understanding_Swim_Lanes.md

> # Contents

Reviewer: ```suggestion
# Scrum swim lanes
```

> 1.0 BACKLOG
> 
> 2.0 TO-DO (CURRENT SPRINT)
> 
> 3.0 IN DEVELOPMENT/PROGRESS
> 
> 4.0 BLOCKED
> 
> 5.0 IN PULL REQUESTS (AWAITING REVIEW)
> 
> 6.0 READY FOR TESTING
> 
> 7.0 IN TESTING
> 
> 8.0 READY FOR REVIEW (TESTING)
> 
> 9.0 DONE

Reviewer: ```suggestion
## Table of contents

* 1.0 Backlog
* 2.0 TO-DO (current sprint)
* 3.0 In development/progress
* 4.0 Blocked
* 5.0 Awaiting review (in pull requests)
* 6.0 Ready for user testing
* 7.0 In user testing
* 8.0 Ready for review (testing)
* 9.0 Done

## Swim lane descriptions
```

> ### 1.0	Backlog

Reviewer: ```suggestion
### 1.0 Backlog
```

> A list of issues created predominantly by the Product Owner (Alan Turning) which are then team review in the backlog refinement session.

Reviewer: ```suggestion
A list of issues created predominantly by the Product Owner (Turing Institute) which are then team review in the backlog refinement session.
```

Note that Alan Turing is a mathematician who died a long time ago. The preferred abbreviation of the Alan Turing Institute is either the Turing Institute or the Turing.

The list of issues stay in the backlog swim lane until the day 1 of sprint planning where thet are then moved to the ‘To-Do’ swim lane

> ### 2.0	To-Do (current sprint)

Reviewer: ```suggestion
### 2.0 To-Do (current sprint)
```

I'm not going to adapt all of these - but please replace tabs with spaces.

Issues prioritised by the Product Owner during backlog refinement sessions and have been agreed to bring into a sprint for work to commence on them will be in this swim lane.
When any given person starts to work on the issue, they must move it to in-development/progress state.

### 3.0	In development/progress
Issues moved from the To-Do state 
Issues in progress during the sprint cycle days 1 to 15

### 4.0	Blocked
A blocker might be an issue or bug which you have come across during development or testing and which is not allowing you to develop or test further
Issues that are marked as blocked may be due to missing information, awaiting information or clarification on the issue.

### 5.0	In pull requests (awaiting review)

> Once branches are considered done (code reviewed) and ready for testing.  A pull request is a way to ask another developer to merge one of your branches into the master branch and that it is ready for testing.

Reviewer: ```suggestion
Once branches are considered done (code reviewed) and ready for testing.
A pull request is a way to ask another developer to merge one of your branches into the master branch and that it is ready for testing.
```
Please put separate sentences on different lines. They will still be rendered as one paragraph so long as there isn't a blank line between them.

### 6.0	Ready for testing
Once issues are considered development complete and code review, approved issues are moved into the Ready for testing state unless it is an issue that does not require testing.
In this instance, you would move the ticket to Done state.

### 7.0	In testing
When you start testing an issue move it into this swim lane.
If you have a concerns whilst testing the issue then add a comment and set it back to the To-Do state.

### 8.0	Ready for review 
When testing is complete and no concerns raised issues will sit under the Ready for review state 
This swim lane is to allow the Product Owner (Alan Turning) to approve the issue and set it to Done.

No concerns means where development tasks have met the success/acceptance criteria as specified in the issue rather than any new customer experience.

> 

Reviewer: Can you define testing? Either in this document or in a separate file. Are you thinking of user testing or technical testing?

Either way I don't think we should be merging anything that is not tested - so maybe step 5.0 could be to open a `[WIP]` (work in progress) pull request and step 8.0 is changing that tag to `[Ready to review]`?

Author: Hi Kirstie,

I have updated section 6.0 to reflect what we mean by testing.
I have updated section 5.0 to reflect two processes. One for documentation review and one for development code review. We have defined two processes, as in an Agile you wouldn't generally add documentation review to a board.
